app/controllers/cma-requirements/other-needs/anything-else-reason.ts

```diff
@@ -0,0 +1,90 @@
+import { NextFunction, Request, Response, Router } from 'express';
+import * as _ from 'lodash';
+import i18n from '../../../../locale/en.json';
+import { paths } from '../../../paths';
+import UpdateAppealService from '../../../service/update-appeal-service';
+import { getConditionalRedirectUrl } from '../../../utils/url-utils';
+import { textAreaValidation } from '../../../utils/validations/fields-validations';
+
+function getAnythingElseReason(req: Request, res: Response, next: NextFunction) {
+  try {
+    const { otherNeeds } = req.session.appeal.cmaRequirements;
+    const savedReason: string = otherNeeds.anythingElseReason;
+
+    return res.render('templates/textarea-question-page.njk', {
+      previousPage: paths.awaitingCmaRequirements.otherNeedsAnythingElse,
+      formAction: paths.awaitingCmaRequirements.otherNeedsAnythingElseReasons,
+      pageTitle: i18n.pages.cmaRequirements.otherNeedsSection.anythingElseReasons.title,
+      question: {
+        name: 'reason',
+        title: i18n.pages.cmaRequirements.otherNeedsSection.anythingElseReasons.heading,
+        value: savedReason ? savedReason : ''
+      },
+      supportingEvidence: false,
+      timeExtensionAllowed: false
+    });
+  } catch (e) {
+    next(e);
+  }
+}
+
+function postAnythingElseReason(updateAppealService: UpdateAppealService) {
+  return async function (req: Request, res: Response, next: NextFunction) {
+    try {
+      const { otherNeeds } = req.session.appeal.cmaRequirements;
+      const savedReason: string = otherNeeds.anythingElseReason;
+
+      const validationErrors = textAreaValidation(req.body['reason'], 'reason', i18n.validationErrors.cmaRequirements.otherNeeds.anythingElseReasonRequired);
+
+      if (validationErrors) {
+        return res.render('templates/textarea-question-page.njk', {
+          previousPage: paths.awaitingCmaRequirements.otherNeedsAnythingElse,
+          formAction: paths.awaitingCmaRequirements.otherNeedsAnythingElseReasons,
+          pageTitle: i18n.pages.cmaRequirements.otherNeedsSection.anythingElseReasons.title,
+          question: {
+            name: 'reason',
+            title: i18n.pages.cmaRequirements.otherNeedsSection.anythingElseReasons.heading,
+            value: savedReason ? savedReason : ''
+          },
+          supportingEvidence: false,
+          timeExtensionAllowed: false,
+          errorList: Object.values(validationErrors),
+          error: validationErrors
+        });
```

Comment:
Not a biggie but could we pull this out into a function we use here and from the get request. Then if we have to add something in the future we don't forget to add it to the model when validation has failed. Was looking at a bug the other day where that had happened and we don't always check the page when we have failed validation so thought that might be a good pattern to use. You'll then have to add the error and errorList. as part of this method I guess.